widget/text.go

```diff
@@ -181,24 +205,26 @@ func (t *textProvider) rowLength(row int) int {
 // CharMinSize returns the average char size to use for internal computation
 func (t *textProvider) charMinSize() fyne.Size {
 	defaultChar := "M"
-	if t.presenter.password() {
+	if t.Presenter().password() {
 		defaultChar = passwordChar
 	}
-	return textMinSize(defaultChar, theme.TextSize(), t.presenter.textStyle())
+	return textMinSize(defaultChar, theme.TextSize(), t.Presenter().textStyle())
 }
 
 // Renderer
 type textRenderer struct {
-	objects []fyne.CanvasObject
-
-	texts []*canvas.Text
-
-	provider *textProvider
+	objects    []fyne.CanvasObject
+	texts      []*canvas.Text
+	provider   *textProvider
+	textsMutex sync.RWMutex
```

Comment:
Is it safe to go with two mutexes for this?  Should there just be a single mutex that locks the whole textRenderer when in use?